### DISCUSSION TITLE
Rename file from inline JS

### DISCUSSION THREAD

**Discussion by sector101010** [chhoumann/quickadd#295] [Q&A]
I have a daily template that uses `inputPrompt` to get the desired date to create the daily note for a specific date.

How can I rename the created file to the value of the `td` variable below? I can ask for the file name in the action settings, but then I have to enter it twice.

```js
const format = 'YYYY-MM-DD'
const input = await this.quickAddApi.inputPrompt('Date of Daily Note to create', format)

let yd, td, tm
if (input) {
    td = moment(input, format)
} else {
    td = moment()
}

yd = moment(td).subtract(1, 'days')
tm = moment(td).add(1, 'days')

return `<< [[${yd.format(format)}]] | [[${td.format(format)}|${td.format('dddd')}]] | [[${tm.format(format)}]] >>`
```

I tried `this.app.workspace.getActiveFile()`, but this just gives me the template file.

Using Obsidian 1.0 with QuickAdd 0.5.5.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

You don't need to rename anything after the fact - the trick is to make the file name and the template share the same variable, so you're only asked once.

If all you need is "ask for a date, use it as the file name and inside the note", modern QuickAdd does that with no script at all: put `{{VDATE:td,YYYY-MM-DD|today}}` in the Template choice's File Name Format and `{{VDATE:td,YYYY-MM-DD}}` in the template body. Same variable name means one prompt shared by both, and it takes natural language ("tomorrow", "next monday", "+7 days").

Your yesterday/today/tomorrow nav line needs date math, so keep a small script - but run it *before* the template in a Macro so it can feed the file name too:

```js
module.exports = async (params) => {
    const format = "YYYY-MM-DD";
    const input = await params.quickAddApi.inputPrompt("Date of daily note", format);
    const td = input ? moment(input, format) : moment();

    params.variables.td = td.format(format);
    params.variables.day = td.format("dddd");
    params.variables.yd = moment(td).subtract(1, "days").format(format);
    params.variables.tm = moment(td).add(1, "days").format(format);
};
```

The Macro is two steps: this user script, then a Nested Choice that runs your Template choice. Set the Template choice's File Name Format to `{{VALUE:td}}`, and put this in the template body:

```
<< [[{{VALUE:yd}}]] | [[{{VALUE:td}}|{{VALUE:day}}]] | [[{{VALUE:tm}}]] >>
```

Variables set in one macro step carry into every later step, so nothing gets typed twice and no rename is needed - the file is created with the right name from the start.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#vdate
